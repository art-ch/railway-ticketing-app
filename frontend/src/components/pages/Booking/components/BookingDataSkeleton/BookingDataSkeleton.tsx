const HeaderSkeleton = () => (
  <div className="h-5 bg-gray-200 rounded animate-pulse w-1/2" />
);

const RowSkeleton = () => (
  <div className="grid grid-cols-[repeat(3,_180px)] gap-2">
    {Array.from({ length: 6 }).map((_, index) => (
      <div key={index} className="h-3 bg-gray-200 rounded animate-pulse" />
    ))}
  </div>
);

const ButtonSkeleton = () => (
  <div className="h-8 bg-gray-200 rounded animate-pulse w-45 space-x-5 ml-auto" />
);

export const BookingDataSkeleton = () => {
  return (
    <div className="flex flex-col space-y-4">
      <HeaderSkeleton />
      <RowSkeleton />
      <HeaderSkeleton />
      <RowSkeleton />
      <ButtonSkeleton />
    </div>
  );
};
