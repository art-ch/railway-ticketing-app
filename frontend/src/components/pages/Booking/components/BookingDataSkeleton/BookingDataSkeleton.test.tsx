import { render } from '@testing-library/react';
import { BookingDataSkeleton } from './BookingDataSkeleton';

describe('BookingDataSkeleton component', () => {
  it('should be rendered correctly', () => {
    const { container } = render(<BookingDataSkeleton />);

    expect(container).toMatchSnapshot();
  });
});
