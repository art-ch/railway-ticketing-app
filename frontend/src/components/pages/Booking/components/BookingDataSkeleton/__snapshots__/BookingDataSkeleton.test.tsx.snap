// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingDataSkeleton component should be rendered correctly 1`] = `
<div>
  <div
    class="flex flex-col space-y-4"
  >
    <div
      class="h-5 bg-gray-200 rounded animate-pulse w-1/2"
    />
    <div
      class="grid grid-cols-[repeat(3,_180px)] gap-2"
    >
      <div
        class="h-3 bg-gray-200 rounded animate-pulse"
      />
      <div
        class="h-3 bg-gray-200 rounded animate-pulse"
      />
      <div
        class="h-3 bg-gray-200 rounded animate-pulse"
      />
      <div
        class="h-3 bg-gray-200 rounded animate-pulse"
      />
      <div
        class="h-3 bg-gray-200 rounded animate-pulse"
      />
      <div
        class="h-3 bg-gray-200 rounded animate-pulse"
      />
    </div>
    <div
      class="h-5 bg-gray-200 rounded animate-pulse w-1/2"
    />
    <div
      class="grid grid-cols-[repeat(3,_180px)] gap-2"
    >
      <div
        class="h-3 bg-gray-200 rounded animate-pulse"
      />
      <div
        class="h-3 bg-gray-200 rounded animate-pulse"
      />
      <div
        class="h-3 bg-gray-200 rounded animate-pulse"
      />
      <div
        class="h-3 bg-gray-200 rounded animate-pulse"
      />
      <div
        class="h-3 bg-gray-200 rounded animate-pulse"
      />
      <div
        class="h-3 bg-gray-200 rounded animate-pulse"
      />
    </div>
    <div
      class="h-8 bg-gray-200 rounded animate-pulse w-45 space-x-5 ml-auto"
    />
  </div>
</div>
`;
